Sources/DiagnosticsLogger.swift

```diff
@@ -88,34 +91,34 @@ extension DiagnosticsLogger {
             return nil
         }
 
-        return queue.sync { try? Data(contentsOf: location) }
+        return queue.sync { try? Data(contentsOf: logFileLocation) }
     }
 
-    /// Removes the log file.
+    /// Removes the log file. Should only be used for testing purposes.
     func deleteLogs() throws {
-        guard FileManager.default.fileExists(atPath: location.path) else { return }
-        try? FileManager.default.removeItem(atPath: location.path)
+        guard FileManager.default.fileExists(atPath: logFileLocation.path) else { return }
+        try? FileManager.default.removeItem(atPath: logFileLocation.path)
     }
 
     private func setup() throws {
-        if !FileManager.default.fileExists(atPath: location.path) {
+        if !FileManager.default.fileExists(atPath: logFileLocation.path) {
             try FileManager.default.createDirectory(atPath: FileManager.default.documentsDirectory.path, withIntermediateDirectories: true, attributes: nil)
-            guard FileManager.default.createFile(atPath: location.path, contents: nil, attributes: nil) else {
+            guard FileManager.default.createFile(atPath: logFileLocation.path, contents: nil, attributes: nil) else {
                 assertionFailure("Unable to create the log file")
                 return
             }
         }
 
-        let fileHandle = try FileHandle(forReadingFrom: location)
-        fileHandle.seekToEndOfFile()
-        logSize = Int64(fileHandle.offsetInFile)
+        logFileHandle = try FileHandle(forWritingTo: logFileLocation)
+        logFileHandle!.seekToEndOfFile()
+        logSize = Int64(logFileHandle!.offsetInFile)
         setupPipe()
         isSetup = true
         startNewSession()
     }
 
     internal func startNewSession() {
-        queue.async {
+        queue.async { [unowned self] in
```

Comment:
I really like were using `[unowned self]` when possible! Just to double-check: Here we use `[unowned self]` instead of `[weak self]`, because this is a singleton object, so we don't risk blocks being executed while the diagnostic logger object is already deallocated.

Reply:
Exactly! The logger is always available in the same lifetime as the queue 🙂